app/src/main/java/org/akvo/flow/presentation/geoshape/entities/AreaShape.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2019 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.akvo.flow.presentation.geoshape.entities;
+
+import android.os.Parcel;
+
+import com.mapbox.mapboxsdk.geometry.LatLng;
+
+import java.util.List;
+
+public class AreaShape extends Shape {
+
+    public AreaShape(String featureId, List<ShapePoint> points) {
+        super(featureId, points);
+    }
+
+    public AreaShape(Parcel in) {
+        super(in);
+    }
+
+    public static final Creator<AreaShape> CREATOR = new Creator<AreaShape>() {
+        @Override
+        public AreaShape createFromParcel(Parcel in) {
+            return new AreaShape(in);
+        }
+
+        @Override
+        public AreaShape[] newArray(int size) {
+            return new AreaShape[size];
+        }
+    };
+
+    @Override
+    public void removeSelectedPoint() {
+        super.removeSelectedPoint();
+        List<ShapePoint> points = getPoints();
+        //There can never be 3, as soon as there are 3 we add the first item to close the area
```

Comment:
The comment is not very clear for me.  Meaning if there are 4 points and one is removed to leave 3, then we add which first item to close the area?

Reply:
We need to add an extra point to close the shape. We only add it after 3 points have been added, so after adding
* 1 or 2 points -> invalid shape, user is still adding points, no closing yet
* 3 points -> shape is valid, add the first point (position 0) to close the shape
* 4 points or more -> nothing else needs to be done since the area is already closed. Closed means all points are linked to 2 other points.

Reply:
but I see now the comment does not make sense here, will modify it a bit